scripts/ci/_utils.sh

```diff
@@ -26,6 +26,8 @@ declare -a EXTRA_DOCKER_PROD_BUILD_FLAGS
 export EXTRA_DOCKER_FLAGS
 export EXTRA_DOCKER_PROD_BUILD_FLAGS
 
+declare -a files_to_cleanup_on_exit=()
```

Comment:
This looks odd, but without it it can fail with "unbound variable" on modern versions of bash when nothing is added to the array.

Reply:
I think it will fail on Mac anyway :(. I had problems with that - empty array is treated as an unbound variable on macOS default Bash (they keep on 3.2.57 as it is GPL2 and anything above is GPLv3 which is not acceptable for Apple). Would be worth checking on Mac.

While -u is useful, I found that sometimes I had to disable it when using arrays that might be empty.

Reply:
Or avoid arrays in cases like that. I think I'd stick to just FILE_TO_CLEAN potentially empty rather then files_to clean in plural.

Reply:
```
❯ bash -c 'set -uo pipefail -e; declare -a foo=(); echo ${#foo[@]}'
0
❯ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin19)
Copyright (C) 2007 Free Software Foundation, Inc.
```

```
❯ bash --version
GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

❯ bash -c 'set -uo pipefail -e; declare -a foo=(); echo ${#foo[@]}' 
0
```